src/main/java/ir/sahab/zk/client/ZkClient.java

```
@@ -320,7 +328,14 @@ public void addPersistentNode(String path, byte[] data, boolean createParent)
         } catch (InterruptedException ex) {
             throw ex;
         } catch (Exception ex) {
-            throw new ZkClientException("Failed to add a persistent ZK node in path: " + path, ex);
+            // There is a corner case where Curator throws a ConnectionLoss but actually we added
+            // the ZK node, so when we retry we get NodeExists exception. In here, we check whether
+            // node created or not and ignore error if exists.
```

whether the node is created or not and ignore the exception if the node exists.